libraries/AP_Mission/AP_Containment_Manager.cpp

```diff
@@ -0,0 +1,217 @@
+#include "AP_Containment_Manager.h"
+#include <AP_Logger/AP_Logger.h>
+#include <GCS_MAVLink/GCS.h>
+
+// Constructor
+AP_Containment_Manager::AP_Containment_Manager(AC_WPNav *&wpnav) :
+_wpnav(wpnav)
+{}
+
+void AP_Containment_Manager::init(void)
+{
+    // Reset the time scaler lpf
+    _time_scaler.set_cutoff_frequency(1.0, 10.0); // 1 Hz cutoff filter
+    reset();
+}
+
+// Function for regularly updating the monitor
+void AP_Containment_Manager::update_time_scaler(const Location current_loc)
+{
+    if (_state == State::DISABLE) {
+        return;
+    }
+
+    AP_Mission *mission = AP::mission();
+    if (mission == nullptr) {
+        return;
+    }
+
+    // We only want to run containment checks if the mission is running
+    if (mission->state() != AP_Mission::mission_state::MISSION_RUNNING) {
+        return;
+    }
+
+    // Rate limit to 5 Hz
+    const uint32_t now = AP_HAL::millis();
+    if (now - _last_scaler_update_ms < 200) {
+        return;
+    }
+
+    // Reset state back to the desired state as we may have been in an error state in the last update.
+    _state = _desired_sate;
+
+    // Calculate the normal-planer radius from the target vector of travel
+    // Get the last WP that we passed.
+    const uint16_t prev_wp_index = mission->get_prev_nav_cmd_index();
+
+    if (prev_wp_index == AP_MISSION_CMD_INDEX_NONE || prev_wp_index < AP_MISSION_FIRST_REAL_COMMAND) {
+        // we have hit the end of the mission or dont have a mission
+        // the calculation below will give nonsense, do not run
+        return;
+    }
+
+    AP_Mission::Mission_Command prev_nav_cmd;
+    if (!mission->get_next_nav_cmd(prev_wp_index, prev_nav_cmd)) {
+        _state = State::ERROR;
```

Comment:
If there is a error here it just carries on? The location will be empty.

Reply:
Fixed